src/Contracts/Task.php

```diff
@@ -0,0 +1,197 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Meilisearch\Contracts;
+
+use Meilisearch\Contracts\TaskDetails\DocumentAdditionOrUpdateDetails;
+use Meilisearch\Contracts\TaskDetails\DocumentDeletionDetails;
+use Meilisearch\Contracts\TaskDetails\DocumentEditionDetails;
+use Meilisearch\Contracts\TaskDetails\DumpCreationDetails;
+use Meilisearch\Contracts\TaskDetails\IndexCreationDetails;
+use Meilisearch\Contracts\TaskDetails\IndexDeletionDetails;
+use Meilisearch\Contracts\TaskDetails\IndexSwapDetails;
+use Meilisearch\Contracts\TaskDetails\IndexUpdateDetails;
+use Meilisearch\Contracts\TaskDetails\SettingsUpdateDetails;
+use Meilisearch\Contracts\TaskDetails\TaskCancelationDetails;
+use Meilisearch\Contracts\TaskDetails\TaskDeletionDetails;
+
+final class Task implements \ArrayAccess
+{
+    /**
+     * @param non-negative-int      $taskUid
+     * @param non-empty-string|null $indexUid
+     * @param array<mixed>          $data     Raw data
+     */
+    public function __construct(
+        private readonly int $taskUid,
+        private readonly ?string $indexUid,
+        private readonly TaskStatus $status,
+        private readonly TaskType $type,
+        private readonly \DateTimeImmutable $enqueuedAt,
+        private readonly ?\DateTimeImmutable $startedAt = null,
+        private readonly ?\DateTimeImmutable $finishedAt = null,
+        private readonly ?string $duration = null,
+        private readonly ?int $canceledBy = null,
+        private readonly ?int $batchUid = null,
+        private readonly ?TaskDetails $details = null,
```

Comment:
When we introduce a new task type in the engine does will this code break?

Reply:
Yes, this lib must be prepared to accept new task types/statuses/task details on new meilisearch releases, same as with other features, that @Strift or others adds when meilisearch gets them :)

Reply:
But when I mean "break," will the user really see a parsing error or something like this if the type is not supported?

Reply:
I'm unsure if this is enough when we decided internally to move Meilisearch (the engine) to v1. One of the requirements for the integrations would be to be able to keep updating the engine no matter which version you have in the SDK.

So far, we've been successful with that promise if I'm not mistaken 😅. Would it be possible to handle it in a way we can redirect the code to a non-typed task type? So if a user updates Meilisearch, it won't break for them, and to get the full capabilities from the SDK, it will just require them to later upgrade the SDK, WDYT?

Reply:
But any change in api could break this sdk, so..

Reply:
> One of the requirements for the integrations would be to be able to keep updating the engine no matter which version you have in the SDK.

if that's a requirement, then we can resort back to using strings, which is sad, but that's your decision